language_bindings/rust/rust_build/src/lib.rs

```diff
@@ -0,0 +1,360 @@
+        );
+
+        let mut gen = Command::new(self.gen_exe.to_str().unwrap());
+        gen.args(["-g", self.gen_name.as_str()]);
+        gen.args(["-f", self.gen_name.as_str()]);
+
+        //Todo change to build dir
+        gen.args(["-o", self.rs_output.to_str().unwrap()]);
+        gen.env("LD_LIBRARY_PATH", self.halide_path.join("lib"));
+
+        //Todo add debug
+        if !self.debug {
+            gen.arg(self.target.as_str());
+        } else {
+            let mut temp = String::from(self.target.as_str());
+            temp.push_str("-debug");
+            gen.arg(temp);
+        }
+        gen.output().expect("failed to run")
+    }
+
+    /// Make the Halide runtime
+    ///
+    /// The Halide runtime is required for using Halide and contains the buffer_t and other useful functions
+    pub fn make_runtime(&self) -> Result<()> {
+        println!(
+            "cargo:rustc-link-search=native={}",
+            self.rs_output.to_str().unwrap()
+        );
+        println!("Cmd to run: {:?}", self.gen_exe.to_str().unwrap());
+        let mut cmd = Command::new(self.gen_exe.to_str().unwrap());
+        cmd.args([
+            "-r",
+            "runtime",
+            "-o",
+            self.rs_output.to_str().unwrap(),
+            "target=host",
+        ]);
+        cmd.env("LD_LIBRARY_PATH", self.halide_path.join("lib"));
+        let output = cmd.output().expect("failed to make runtime");
+        let result;
+        if output.status.success() {
+            result = self.rename_runtime();
+            if result.is_ok() {
+                println!("cargo:rustc-link-lib=static={}", "runtime");
+            }
+        } else {
+            let errors =
+                String::from("failed to make runtime with gen: ").add(self.gen_name.as_str());
+            result = Err(io::Error::new(ErrorKind::Other, errors.as_str()));
+        }
+        result
+    }
+
+    fn rename_runtime(&self) -> Result<()> {
+        println!(
+            "file name: {:?}",
+            self.rs_output
+                .join("runtime")
+                .with_extension("a")
+                .to_str()
+                .unwrap()
+        );
+        println!(
+            "file out: {:?}",
+            self.rs_output
+                .join(String::from("lib").add("runtime"))
+                .with_extension("a")
+                .to_str()
+                .unwrap()
+        );
+        fs::rename(
+            self.rs_output
+                .join("runtime")
+                .with_extension("a")
+                .to_str()
+                .unwrap(),
+            self.rs_output
+                .join(String::from("lib").add("runtime"))
+                .with_extension("a")
+                .to_str()
+                .unwrap(),
+        )
+    }
+
+    /// Renames the gen outputs to be what rust and bindgen expect on linux distros
+    ///
+    /// IE halide.a -> libhalide.a
+    pub fn rename(&self) -> Result<()> {
```

Comment:
This seems like an implementation detail. Does it really need to be public? Further, is there an underlying issue that should be fixed in Halide's build system rather than hacked around here?

Reply:
No, static libraries don't get a `lib` prefix on Linux/Unix.

Reply:
Not with the CMake build system, but maybe with GenGen.cpp's output? Our releases include `libHalide.a`, for example.

Reply:
I'm a bit confused about whether this code is dealing with the Halide runtime or with compiled Halide code. If the latter, we should avoid creating a name collision with the runtime library.